test/integration/TypeBrowserTest.kt

```diff
@@ -0,0 +1,124 @@
+
+package com.vaticle.typedb.studio.test.integration
+
+
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onAllNodesWithText
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import com.vaticle.typedb.studio.state.StudioState
+import org.junit.Ignore
+import org.junit.Rule
+import org.junit.Test
+import java.util.*
+
+
+class TypeBrowserTest {
+    @get:Rule
+    val composeRule = createComposeRule()
+
+    @Test
+    fun interactiveSchemaWritesAutomaticallyDisplayed() {
+        val uuid = UUID.randomUUID().toString()
+        studioTest(composeRule) {
+            // We have to open a project to enable the '+' to create a new file.
+            cloneAndOpenProject(composeRule, TQL_DATA_PATH, uuid)
+            connectToTypeDB(composeRule, DB_ADDRESS)
+            createDatabase(composeRule, uuid)
+            writeSchemaInteractively(composeRule, uuid, SCHEMA_FILE_NAME)
+
+            // We can assert that the schema has been written successfully here as the schema
+            // is shown in the type browser.
+            composeRule.onNodeWithText("attribute").assertExists()
+            composeRule.onNodeWithText("commit-date").assertExists()
+            composeRule.onNodeWithText("commit-hash").assertExists()
+        }
+    }
+
+    @Test
+    fun collapseTypes() {
+        val uuid = UUID.randomUUID().toString()
+        studioTest(composeRule) {
+            connectToTypeDB(composeRule, DB_ADDRESS)
+            cloneAndOpenProject(composeRule, TQL_DATA_PATH, uuid)
+            createDatabase(composeRule, uuid)
+            writeSchemaInteractively(composeRule, uuid, SCHEMA_FILE_NAME)
+
+            composeRule.onAllNodesWithText("Project").get(0).performClick()
+            composeRule.onAllNodesWithText("Project").get(1).performClick()
+            wait(composeRule, 500)
+
+            composeRule.onNodeWithText(DOUBLE_CHEVRON_UP_ICON_STRING).performClick()
+            wait(composeRule, 500)
+
+            composeRule.onNodeWithText("commit-date").assertDoesNotExist()
+        }
+    }
+
+    @Test
+    fun expandTypes() {
+        val uuid = UUID.randomUUID().toString()
+        studioTest(composeRule) {
+            connectToTypeDB(composeRule, DB_ADDRESS)
+            cloneAndOpenProject(composeRule, TQL_DATA_PATH, uuid)
+            createDatabase(composeRule, uuid)
+            writeSchemaInteractively(composeRule, uuid, SCHEMA_FILE_NAME)
+
+            composeRule.onNodeWithText(DOUBLE_CHEVRON_UP_ICON_STRING).performClick()
+            wait(composeRule, 500)
+
+            composeRule.onNodeWithText("commit-date").assertDoesNotExist()
+
+            composeRule.onNodeWithText(DOUBLE_CHEVRON_DOWN_ICON_STRING).performClick()
+            wait(composeRule, 500)
+
+            composeRule.onNodeWithText("commit-date").assertExists()
+        }
+    }
+
+    // This test is ignored as the export schema button doesn't open a new file as expected.
```

Comment:
Is there an issue logged for this bug?

Reply:
This comment wasn't clear enough - the functionality works when actually using Studio but not within the test framework. I've reworded it as such.